include/boost/astronomy/coordinate/arithmetic.hpp

```diff
@@ -72,54 +72,161 @@ auto cross
         bg::cs::cartesian
     > tempPoint1, tempPoint2, result;
 
+    auto cartesian1 = make_cartesian_representation(representation1);
```

Comment:
There is no need to convert `representation1` to cartesian as it is already in cartesian.